lib/json_schema/parser.rb

```diff
@@ -16,6 +16,8 @@ class Parser
       String     => "string",
       TrueClass  => "boolean",
     }
+    EMPTY_ARRAY = [].freeze
```

Comment:
Would you mind adding a brief comment over these that explains they're an optimization to reduce the number of empty arrays and hashes that are allocated? I'm a little worried that it's a bit non-obvious to laypeople (including me if I was reading this without context), which puts us in some danger of a regression. Thanks!

Reply:
done!